backend/dataall/modules/metadata_forms/services/metadata_form_service.py

```diff
@@ -0,0 +1,74 @@
+from dataall.base.context import get_context
+from dataall.base.db import exceptions, paginate
+from dataall.core.organizations.db.organization_repositories import OrganizationRepository
+from dataall.core.environment.db.environment_repositories import EnvironmentRepository
+
+from dataall.modules.metadata_forms.db.enums import MetadataFormVisibility
+from dataall.modules.metadata_forms.db.metadata_form_repository import MetadataFormRepository
+
+
+class MetadataFormParamValidationService:
+    @staticmethod
+    def validate_create_form_params(data):
+        visibility = data.get('visibility', MetadataFormVisibility.Team.value)
+        if not MetadataFormVisibility.has_value(visibility):
+            data['visibility'] = MetadataFormVisibility.Team.value
+
+        if 'SamlGroupName' not in data:
+            raise exceptions.RequiredParameter('SamlGroupName')
+
+        if 'homeEntity' not in data and (
+            visibility == MetadataFormVisibility.Organization.value
+            or visibility == MetadataFormVisibility.Environment.value
+        ):
+            raise exceptions.RequiredParameter('homeEntity')
+
+        if 'name' not in data:
+            data['name'] = 'New Form'
```

Comment:
I would think we should raise error if no name is provided rather than set a default of `New Form` 

 ++ also here let's check the value of `name` key - as of now I can create a form with no name and does not default to `New Form`

Reply:
[aea1745](https://github.com/data-dot-all/dataall/pull/1444/commits/aea1745ee8adcd027cb019f967e21c5c3a4849ca)